src/java.base/share/classes/java/util/concurrent/Executors.java

```diff
@@ -824,17 +825,6 @@ public <T> T invokeAny(Collection<? extends Callable<T>> tasks,
         }
     }
 
-    private static class FinalizableDelegatedExecutorService
-            extends DelegatedExecutorService {
-        FinalizableDelegatedExecutorService(ExecutorService executor) {
-            super(executor);
-        }
-        @SuppressWarnings("removal")
-        protected void finalize() {
-            super.shutdown();
-        }
-    }
-
```

Comment:
You might consider keeping a dedicated subclass ("CleanableDelegatedExecutorService"?). Such a class could save the Cleanable from Cleaner.register(), and override the shutdown() method to call Cleanable.clean(). This would reduce GC reference tracking, for instance in places where newSingleThreadExecutor() is used in a try-with-resources.

Reply:
That might be better as that would clear the reference (so it won't be queued) when explicit shutdown (or closed). We do that in several places (as you know) at the cost of a reference to the Cleanable allowing "this" to escape during construction.